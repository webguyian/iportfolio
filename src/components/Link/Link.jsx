import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { Link as RouterLink } from 'react-router-dom';

import Icon from 'components/Icon/Icon';

const Link = forwardRef((props, ref) => {
  const { back, backLabel, children, className, to, ...otherProps } = props;
  const baseClass = 'ui-link';
  const backLinkClass = `${baseClass}--back`;

  if (back) {
    return (
      <RouterLink
        className={classNames(baseClass, backLinkClass, className)}
        to={to}
        ref={ref}
        {...otherProps}
      >
        <Icon name="chevron-left" /> {backLabel}
      </RouterLink>
    );
  }

  return (
    <RouterLink
      className={classNames(baseClass, className)}
      to={to}
      ref={ref}
      {...otherProps}
    >
      {children}
    </RouterLink>
  );
});

Link.displayName = 'Link';

Link.propTypes = {
  back: PropTypes.bool,
  backLabel: PropTypes.string,
  children: PropTypes.node,
  className: PropTypes.string,
  to: PropTypes.oneOfType([PropTypes.object, PropTypes.string])
};

Link.defaultProps = {
  back: false,
  backLabel: 'Back'
};

export default Link;
