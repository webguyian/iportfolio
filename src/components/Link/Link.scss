.ui-link {
  color: $white;
  text-decoration: none;

  &--back {
    margin-right: auto;
    color: $yellow;
  }
}
