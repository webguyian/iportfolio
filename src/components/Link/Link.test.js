import React from 'react';
import { create } from 'react-test-renderer';

import Link from './Link';

jest.unmock('components/Link/Link');

describe('<Link />', () => {
  it('renders correctly', () => {
    const component = create(<Link to="/">Go home</Link>);

    expect(component).toMatchSnapshot();
  });

  it('renders correctly as back link', () => {
    const component = create(<Link to="/" back />);

    expect(component).toMatchSnapshot();
  });
});
