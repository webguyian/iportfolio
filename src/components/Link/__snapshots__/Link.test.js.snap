// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> renders correctly 1`] = `
<Link
  className="ui-link"
  to="/"
>
  Go home
</Link>
`;

exports[`<Link /> renders correctly as back link 1`] = `
<Link
  className="ui-link ui-link--back"
  to="/"
>
  <i
    className="ui-icon ui-icon-chevron-left fas fa-chevron-left"
  />
   
  Back
</Link>
`;
